src/MessageWriter.ts

```diff
@@ -203,6 +213,15 @@ export class MessageWriter {
           writer.sequenceLength(arrayLength);
         }
 
+        if (field.arrayLength != undefined && nestedMessage != undefined) {
+          const givenFieldLength = fieldLength(nestedMessage);
+          if (givenFieldLength !== field.arrayLength) {
+            throw new Error(
+              `Given array does not match the defined array size: ${givenFieldLength} != ${field.arrayLength}`,
```

Comment:
wording suggestion: `` `Expected ${field.arrayLength} items for fixed-length array field ${field.name} but received ${givenFieldLength}` `` ... or something like that